requests/models.py

```diff
@@ -796,7 +800,24 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             if delimiter:
                 lines = chunk.split(delimiter)
             else:
+                # Python splitlines() supports the universal newline (PEP 278).
+                # That means, '\r', '\n', and '\r\n' are all treated as end of
+                # line. If the last chunk ends with '\r', and the current chunk
+                # starts with '\n', they should be merged and treated as only
+                # *one* new line separator '\r\n' by splitlines().
+                # This rule only applies when splitlines() is used.
+                if last_chunk_ends_with_cr and chunk.startswith(line_feed):
```

Comment:
Yep, I'm on board with @sigmavirus24's suggestion. That seems the most readable option out of what's been suggested.

The only minor input I have is standardizing the use of `_` in `ends_with` vs `startswith` in the variable names.